machine/machine_core/ssh_connection.py

```diff
@@ -437,17 +437,21 @@ def download_dir(self, source, dest, relative_dir="."):
         except:
             self.message("Error while downloading directory '{0}'".format(source))
 
-    def write(self, dest, content):
+    def write(self, dest, content, append=False, mode=None):
         """Write a file into the test machine
 
         Arguments:
             content: Raw data to write to file
             dest: The file name in the machine to write to
+            append: If True, append to existing file instead of replacing it
+            mode: Optional file access mode as shell string (e.g. "0600")
```

Comment:
`append` and `chmod` together is strange to imagine...

Reply:
True that, same for append+owner, but *technically* it works.